easybuild/easyconfigs/f/FHI-aims/FHI-aims-240507-intel-2023a.eb

```
@@ -0,0 +1,57 @@
+##
+# This file is an EasyBuild reciPY as per https://github.com/easybuilders/easybuild
+#
+# Authors::   Dugan Witherick (University of Warwick)
+# Updated to 240507 
+# J. Sassmannshausen (Imperial College London/UK)
+##
+
+easyblock = 'CMakeMake'
+
+name = 'FHI-aims'
+version = '240507'
+
+homepage = 'https://fhi-aims.org/'
+description = """FHI-aims is an efficient, accurate all-electron,
+full-potential electronic structure code package for computational molecular  
+and materials science (non-periodic and periodic systems). The code supports  
+DFT (semilocal and hybrid) and many-body perturbation theory. FHI-aims is     
+particularly efficient for molecular systems and nanostructures, while        
+maintaining high numerical accuracy for all production tasks. Production      
+calculations handle up to several thousand atoms and can efficiently use (ten)thousands of cores.
+"""
+
+toolchain = {'name': 'intel', 'version': '2023a'}
+toolchainopts = {'opt': True, 'precise': True}
+
+download_instructions = """
+The source code must be downloaded manually from the FHI-aims club
+(https://fhi-aims.org/get-the-code-menu/login).
+Access to the FHI-aims club requires a valid license and registration.
+Details on available license options and how to register to access
+FHI-aims club may be found at:
+https://fhi-aims.org/get-the-code-menu/get-the-code """
+
+sources = ['%(namelower)s.%(version)s.tar.gz']
+checksums = ['dea6483834d80ef5d8826171bd710ebcb035c4b971c4c958c82fdb9f03af6a6b']
+
+builddependencies = [('CMake', '3.26.3')]
+
+configopts = ' -DCMAKE_Fortran_COMPILER="$MPIF90" -DCMAKE_CXX_COMPILER="icpc"  -DCMAKE_C_COMPILER="icc" '
```

To honor toolchain options `oneapi` and `usempi`:

```suggestion
configopts = ' -DCMAKE_Fortran_COMPILER="$FC" -DCMAKE_CXX_COMPILER="$CXX_SEQ"  -DCMAKE_C_COMPILER="$CC_SEQ" '
```

According to https://fhi-aims-club.gitlab.io/tutorials/cmake-tutorial/cmake_tutorial they need an MPI fortran compiler --> `'usempi': True` in toolchainopts

---

Thanks for that! I done a bit more digging and with the help of the FHI-aims team I think I got it all sorted so we can use `oneAPI` as the compiler now. From what I understand, there are only a few places where `MPI` is being used and thus when you link it is important to use `MPI` but not when compiling. I got told that eliminates problems with wrapper-scripts. Nonetheless, I used `MPI` for all three languages. Let me know what you think of the improvements done.